### PR TITLE
Alpha deploy override flags

### PR DESCRIPTION
**Pull request by skhalash**
**Description**

Changes proposed in this pull request:

Implement the following CLI flags:
- Domain, TLSCrt, TLSKey
- Profile
- Value
- ValueFile

Also, migrate the `download` pkg from hydroform/parallel-install (renamed it to `resolve`).

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/12039

### CODE REVIEW

**Review by NHingerl — DISMISSED** [review]

File: cmd/kyma/alpha/deploy/values.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+			return nil
+		}
+		return errors.Wrap(err, "failed to add default values file")
+	}
+
+	return nil
+}
+
+func addValueFiles(builder *overrides.Builder, opts *Options, workspace *workspace.Workspace) error {
+	valueFiles, err := resolve.Files(opts.ValueFiles, filepath.Join(workspace.WorkspaceDir, "tmp"))
```
Comment (rakesh-garimella) [inline cmd/kyma/alpha/deploy/values.go:59]: what does `tmp` do here ?

**Review by rakesh-garimella — APPROVED** [review]

File: docs/gen-docs/kyma_alpha_deploy.md [hunk 2]
```diff
@@ -17,6 +17,12 @@ kyma alpha deploy [flags]
 ```bash
       --component strings        Provide one or more components to deploy (e.g. --component componentName@namespace)
   -c, --components-file string   Path to the components file (default "$HOME/.kyma/sources/installation/resources/components.yaml" or ".kyma-sources/installation/resources/components.yaml")
+  -d, --domain string            Custom domain used for installation
+  -p, --profile string           Kyma deployment profile. If not specified, Kyma uses its default configuration. The supported profiles are: evaluation, production
+      --tls-crt string           TLS certificate file for the domain used for installation
+      --tls-key string           TLS key file for the domain used for installation
+      --value strings            Set configuration values. Can specify one or more values, also as a comma-separated list (e.g. --value component.a='1' --value component.b='2' or --value component.a='1',component.b='2').
+  -f, --values-file strings      Path(s) to one or more JSON or YAML files with configuration values
```
Comment (majakurcius) [inline docs/gen-docs/kyma_alpha_deploy.md:25]:
```suggestion
  -d, --domain string            Custom domain used for installation.
  -p, --profile string           Kyma deployment profile. If not specified, Kyma uses its default configuration. The supported profiles are: evaluation, production.
      --tls-crt string           TLS certificate file for the domain used for installation.
      --tls-key string           TLS key file for the domain used for installation.
      --value strings            Set configuration values. Can specify one or more values, also as a comma-separated list (e.g. --value component.a='1' --value component.b='2' or --value component.a='1',component.b='2').
  -f, --values-file strings      Path(s) to one or more JSON or YAML files with configuration values.
```

File: cmd/kyma/alpha/deploy/cmd.go [hunk 3]
```diff
@@ -54,6 +51,15 @@ func NewCmd(o *Options) *cobra.Command {
...
+
+	cobraCmd.Flags().StringVarP(&o.Domain, "domain", "d", "", "Custom domain used for installation")
+	cobraCmd.Flags().StringVarP(&o.Profile, "profile", "p", "",
+		fmt.Sprintf("Kyma deployment profile. If not specified, Kyma uses its default configuration. The supported profiles are: %s, %s", profileEvaluation, profileProduction))
+	cobraCmd.Flags().StringVarP(&o.TLSCrtFile, "tls-crt", "", "", "TLS certificate file for the domain used for installation")
+	cobraCmd.Flags().StringVarP(&o.TLSKeyFile, "tls-key", "", "", "TLS key file for the domain used for installation")
+	cobraCmd.Flags().StringSliceVarP(&o.Values, "value", "", []string{}, "Set configuration values. Can specify one or more values, also as a comma-separated list (e.g. --value component.a='1' --value component.b='2' or --value component.a='1',component.b='2').")
+	cobraCmd.Flags().StringSliceVarP(&o.ValueFiles, "values-file", "f", []string{}, "Path(s) to one or more JSON or YAML files with configuration values")
```
Comment (majakurcius) [inline cmd/kyma/alpha/deploy/cmd.go:61]:
```suggestion
	cobraCmd.Flags().StringVarP(&o.Domain, "domain", "d", "", "Custom domain used for installation.")
	cobraCmd.Flags().StringVarP(&o.Profile, "profile", "p", "",
		fmt.Sprintf("Kyma deployment profile. If not specified, Kyma uses its default configuration. The supported profiles are: %s, %s.", profileEvaluation, profileProduction))
	cobraCmd.Flags().StringVarP(&o.TLSCrtFile, "tls-crt", "", "", "TLS certificate file for the domain used for installation.")
	cobraCmd.Flags().StringVarP(&o.TLSKeyFile, "tls-key", "", "", "TLS key file for the domain used for installation.")
	cobraCmd.Flags().StringSliceVarP(&o.Values, "value", "", []string{}, "Set configuration values. Can specify one or more values, also as a comma-separated list (e.g. --value component.a='1' --value component.b='2' or --value component.a='1',component.b='2').")
	cobraCmd.Flags().StringSliceVarP(&o.ValueFiles, "values-file", "f", []string{}, "Path(s) to one or more JSON or YAML files with configuration values.")
```

**Review by majakurcius — COMMENTED** [review]
I added two comments to make the commands follow the same pattern. (Some of them consisted of a few sentences. It looked weird not having this last sentence end with a period when it should but didn't 'cause others didn't too, therefore I added the period to all sentences.)

**Review by rakesh-garimella — APPROVED** [review]

**Review by majakurcius — APPROVED** [review]
🚀

**Pull request merged**